### PR TITLE
[task] return error if aws or az cli are not installed

### PR DESCRIPTION
**Pull request by pducolin**
What does this PR do?
---------------------

Add check at setup time to ensure az and aws cli are installed

Which scenarios this will impact?
-------------------

None

Motivation
----------

I did not have az cli installed, don't think it comes by default on our laptops. Better to warn contributors inline.

Additional Notes
----------------

### CODE REVIEW

File: tasks/setup.py [hunk 1]
```diff
@@ -26,6 +26,17 @@ def setup(
     """
     Setup a local environment, interactively by default
     """
+    # Ensure aws cli is installed
+    result = ctx.run("aws --version", warn=True, hide=True)
```
Comment (FlorentClarret) [inline tasks/setup.py:30]: You could use [shutil.which](https://docs.python.org/3.11/library/shutil.html#shutil.which) (works on Windows too) with `if not shutil.which("aws"):`

**Review by FlorentClarret — APPROVED** [review]
LGTM but left one suggestion!

File: tasks/setup.py (hunk 1 above)
Comment (pducolin) [reply tasks/setup.py:30]: Ah, thank you!!

Comment (pducolin) [reply tasks/setup.py]: Updated

**Review by FlorentClarret — APPROVED** [review]

File: tasks/setup.py (hunk 1 above)
Comment (FlorentClarret) [reply tasks/setup.py]: Nice!

**Pull request merged**